docs/website/docs/command-reference/init.md

```diff
@@ -0,0 +1,99 @@
+---
+title: odo init
+sidebar_position: 6
+---
+
+The `odo init` command is the first command to be executed when you want to bootstrap a new development, using `odo`. If sources already exist,
+the command `odo dev` should be considered instead.
+
+This command must be executed from an empty directory, and as a result, the command will download a `devfile.yaml` file and, optionally, a starter project.
+
+The command can be exectued in two flavors, either interactive or non-interactive.
+
+## Interactive mode
+
+In interactive mode, you will be guided to choose a devfile from the list of devfiles present in the registry or registries referenced (using the `odo registry` command) and a starter project referenced by the selected devfile, and you will be asked for a name for the component present in the devfile.
+
+## Non-interactive mode
+
+In non-interactive mode, you will have to specify from the command-line the information needed to get a devfile.
+
+If you want to download a devfile from a registry, you must specify the devfile name with the `--devfile` flag. The devfile with the specified name will be searched into the registries referenced (using `odo registry`), and the first one matching will be downloaded. If you want to download the devfile from a specific registry in the list or referenced registries, you can use the `--devfile-registry` flag to specify the name of this registry.
+
+If you prefer to download a devfile from an URL or from the local filesystem, you can use the `--devfile-path` instead.
+
+The `--starter` flag indicates the name of the starter project (as referenced in the selected devfile), that you want to use to start your development.
```

Comment:
```suggestion
The `--starter` flag indicates the name of the starter project (as referenced in the selected devfile), that you want to use to start your development. To see the available starter projects for a component type, use `odo catalog describe component <component-type>`.
```

I personally don't like the `component-type` or "component type" in the above suggestion, so if you could find something better, use that.